app/controllers/public_images_controller.rb

```diff
@@ -56,7 +56,11 @@ def destroy
     if params[:page].present?
       redirect_to edit_bike_url(imageable_id, page: params[:page]) and return
     elsif imageable_type == "Blog"
-      redirect_to edit_admin_news_url(@imageable.title_slug) and return
+      if request.xhr?
+        render json: {success: true} and return
```

Comment:
Why is this returning something different from the json returned above? Why check for xhr instead of json format?

Reply:
mmm, because before I added this my page was, for some reason trying to hit a weird route. It would delete the image and then instead of redirecting the way I wanted it to, it would redirect to DELETE `admin/news/"slug"`. So instead of hitting the regular redirect I just have it removed on the front with JS if it was JUST added.